### PR TITLE
Sanitize filenames in windows

### PR DESCRIPTION
**Pull request by de-odex**

### CODE REVIEW

File: slider/library.py [hunk 1]
```diff
@@ -259,13 +261,19 @@ def save(self, data, *, beatmap=None):
...
-        path = self.path / (
+        beatmap_file_name = (
             f'{beatmap.artist} - '
             f'{beatmap.title} '
             f'({beatmap.creator})'
             f'[{beatmap.version}]'
             f'.osu'
         ).replace('/', '_')
+        if _platform.startswith("win"): beatmap_file_name = _sub(
+            '[:*?"<>|]',
```
Comment (llllllllll) [inline slider/library.py]: this is interpreted as a regex and everything but `:<>` are special in regex. Maybe we should just use a little loop and `str.replace`

File: slider/library.py [hunk 2]
```diff
@@ -4,6 +4,8 @@
 import os
 import pathlib
 from time import sleep
+from sys import platform as _platform
+from re import sub as _sub
```
Comment (llllllllll) [inline slider/library.py]: why are we aliasing the imports, if we want to namespace these we can just do `import sys` and `import re` and use the qualified names later.

File: slider/library.py [hunk 3]
```diff
@@ -259,13 +261,19 @@ def save(self, data, *, beatmap=None):
...
 
-        path = self.path / (
+        beatmap_file_name = (
             f'{beatmap.artist} - '
             f'{beatmap.title} '
             f'({beatmap.creator})'
             f'[{beatmap.version}]'
             f'.osu'
         ).replace('/', '_')
+        if _platform.startswith("win"): beatmap_file_name = _sub(
```
Comment (llllllllll) [inline slider/library.py]: it is bad practice to put the statement on the same line as the condition, whitespace is cheap so you should just put this on a new line.

**Comment by llllllllll** [comment]
Thanks for fixing this!

File: slider/library.py (hunk 1 above)
Comment (llllllllll) [inline slider/library.py]: also we are missing `\`

Comment (de-odex) [inline slider/library.py]: so am i replacing it with a loop?

**Pull request merged**